kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/partition/DynamicPartitioner.java

```diff
@@ -0,0 +1,256 @@
+                               Set<Schema> schemas) throws BigQueryConnectException,
+                                                           InterruptedException {
+    logger.debug("Seeking best batch size...");
+    int currentIndex = 0;
+    int successfulCallCount = 0;
+    while (currentIndex < elements.size()) {
+      int endIndex = currentIndex + currentBatchSize;
+      List<InsertAllRequest.RowToInsert> currentPartition =
+          elements.subList(currentIndex, endIndex);
+      try {
+        writer.writeRows(table, currentPartition, topic, schemas);
+        // success
+        successfulCallCount++;
+        currentIndex = endIndex;
+        // if our batch size is the maximum batch size we are done finding a batch size.
+        if (currentBatchSize == MAXIMUM_BATCH_SIZE) {
+          seeking = false; // case 3
+          logger.debug("Best batch size found (max): {}", currentBatchSize);
+          establishedWriteAll(table,
+                              elements.subList(currentIndex, elements.size()),
+                              topic,
+                              schemas);
+          return;
+        }
+        //  increase the batch size if: we actually did just test that batchSize
+        if (!(currentPartition.size() < currentBatchSize)) {
+          increaseBatchSize();
+        }
+      } catch (BigQueryException exception) {
+        // failure
+        if (isPartitioningError(exception)) {
+          decreaseBatchSize();
+          // if we've had at least 1 successful call we'll assume this is a good batch size.
+          if (successfulCallCount > 0) {
+            seeking = false; // case 1
+            logger.debug("Best batch size found (error if higher): {}", currentBatchSize);
+            establishedWriteAll(table,
+                                elements.subList(currentIndex, elements.size()),
+                                topic,
+                                schemas);
+            return;
+          }
+        } else {
+          throw new BigQueryConnectException(
+              String.format("Failed to write to BigQuery table %s", table),
+              exception);
+        }
+      }
+    }
+    // if we finished all this with only one successful call...
+    if (successfulCallCount ==  1) {
+      // then we are done finding a batch size.
+      seeking = false; // case 2
+      // decrease batch size back to the size that we actually checked.
+      decreaseBatchSize();
+      logger.debug("Best batch size found (all elements): {}", currentBatchSize);
+    }
+  }
+
+  /**
+   * writeAll intended for a generally stable currentBatchSize.
+   *
+   * <p>Write all the given elements to BigQuery in the smallest possible number of batches, as
+   * evenly as possible.
+   *
+   * <p>If we encounter a partitioning-related error, we will immediately lower the batch size and
+   * try again.
+   *
+   * <p>Every {@link #CONT_SUCCESS_COUNT_BUMP} calls, if there have been no errors, we will bump up
```

Comment:
I mean cont as in continuous (how many successes in a row before we bump the batchSize up)